toolz/functoolz.py

```diff
@@ -681,3 +659,172 @@ def __name__(self):
             return '%s_excepting_%s' % (self.f.__name__, exc_name)
         except AttributeError:
             return 'excepting'
+
+
+def is_valid_args(func, args, kwargs):
```

Comment:
We can.  I would like to do what is most clear to the user.  Avoiding an `if` statement or two is not a compelling argument imo.  Calling curried objects should be as fast as possible, but returning new curried objects may be relatively slow.  Similarly, I doubt that `is_valid_args` is expected to be fast.

The advantage of including all branches in a single function is to be upfront about differences in Python versions when inspecting the function in a REPL.

I don't have a strong opinion here.  I'll see if I can make it cleaner by doing what you suggest.